include/envoy/http/codec.h

```diff
@@ -337,7 +339,7 @@ class Connection {
    * Dispatch incoming connection data.
    * @param data supplies the data to dispatch. The codec will drain as many bytes as it processes.
    */
-  virtual void dispatch(Buffer::Instance& data) PURE;
+  virtual absl::Status dispatch(Buffer::Instance& data) PURE;
```

Comment:
Can virtual functions be  marked with ABSL_MUST_USE_RESULT?

Reply:
I think this suffices: `absl::Status` is marked as ABSL_MUST_USE_RESULT in the absl library, so I do get compiler errors when I don't use the return value of `dispatch(_)` now. 
(for eg
`error: ignoring return value of function declared with 'warn_unused_result' attribute [-Werror,-Wunused-result]
        codec_->dispatch(buffer);`)